mp/src/game/client/momentum/ui/leaderboards/LeaderboardsTimes.cpp

```diff
@@ -358,34 +361,33 @@ void CLeaderboardsTimes::LoadLocalTimes(KeyValues* kv)
         V_FixSlashes(path);
 
         FileFindHandle_t found;
+        CUtlStringList filenames;
         const char *pFoundFile = filesystem->FindFirstEx(path, "MOD", &found);
         while (pFoundFile)
         {
             // NOTE: THIS NEEDS TO BE MANUALLY CLEANED UP!
             char pReplayPath[MAX_PATH];
             V_ComposeFileName(RECORDING_PATH, pFoundFile, pReplayPath, MAX_PATH);
 
-            CMomReplayBase *pBase = g_ReplayFactory.LoadReplayFile(pReplayPath, false);
-            Assert(pBase != nullptr);
-
-            if (pBase)
-                m_vLocalTimes.InsertNoSort(pBase);
+            filenames.CopyAndAddToTail(pReplayPath);
 
             pFoundFile = filesystem->FindNext(found);
         }
 
         filesystem->FindClose(found);
 
-        if (!m_vLocalTimes.IsEmpty())
-        {
-            m_vLocalTimes.RedoSort();
-            m_bTimesNeedUpdate[TIMES_LOCAL] = false;
-        }
+        auto callback = UtlMakeDelegate(this, &CLeaderboardsTimes::LocalTimesCallback);
+        g_ReplayFactory.AsyncLoadReplayFileMultiple(filenames, "MOD", callback, &m_mapReplayLocalRequest);
```

Comment:
Callback variable isn't used anywhere so you can just do this:
```suggestion
        g_ReplayFactory.AsyncLoadReplayFileMultiple(filenames, "MOD", UtlMakeDelegate(this, &CLeaderboardsTimes::LocalTimesCallback), &m_mapReplayLocalRequest);
```